gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/ModuleFacetUtil.java

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.gradle.core;
+
+import com.liferay.ide.core.util.FileUtil;
+
+import java.io.File;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+
+/**
+ * @author Andy Wu
+ */
+public class ModuleFacetUtil
+{
+
+    public static void addFacets( final IProject project, IProgressMonitor monitor ) throws CoreException
+    {
+        addNature( project, "org.eclipse.wst.common.modulecore.ModuleCoreNature", monitor );
+        addNature( project, "org.eclipse.wst.common.project.facet.core.nature", monitor );
```

Comment:
Can you try this without adding the "org.eclipse.wst.common.project.facet.core.nature" nature?  I would like to avoid the faceted settings if possible.